File: contains_duplicate.rb
# @param {Integer[]} nums
# @return {Boolean}
def contains_duplicate(nums)
  nums_occurences = {}

  nums.each do |num|
    return true if nums_occurences[num]

    nums_occurences[num] = 1
  end
  false
end

nums = [-1, 0, -1]
puts contains_duplicate(nums)

File: contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        numCount = {}
        for num in nums:
          if num in numCount:
            return True
          numCount[num] = 0

        return False

nums = [1,1,1,3,3,4,3,2,4,2]
test = Solution().containsDuplicate(nums)
print(test)
